server/container/guice/memory-guice/src/test/java/org/apache/james/GuiceJamesServerTest.java

```diff
@@ -99,5 +197,24 @@ void serverShouldNotBeStartedOnUncaughtException(GuiceJamesServer server) throws
 
             assertThat(server.isStarted()).isFalse();
         }
+
+        @Test
+        void serverShouldThrowWhenAStartUpCheckFails(GuiceJamesServer server) throws Exception {
+            serverWithStartUpCheck = server.overrideWith(binder -> {
+                Multibinder<StartUpChecksPerformer.StartUpCheck> setBinder = Multibinder
+                    .newSetBinder(binder, StartUpChecksPerformer.StartUpCheck.class);
+
+                setBinder.addBinding().to(NoopStartUpCheck.class);
+                setBinder.addBinding().to(FailingStartUpCheck.class);
+            });
+
+            assertThatThrownBy(serverWithStartUpCheck::start)
+                .isInstanceOf(StartUpChecksPerformer.StartUpChecksException.class)
```

Comment:
I don't think you want to check the exception message.
Exceptions can carry information, please use that to check the member objects instead of the message